src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -76,113 +76,104 @@ const unsigned mode_flag_custom = 1;
-	    _system_type == MAV_TYPE_HEXAROTOR ||
-	    _system_type == MAV_TYPE_OCTOROTOR ||
-	    _system_type == MAV_TYPE_VTOL_DUOROTOR ||
-	    _system_type == MAV_TYPE_VTOL_QUADROTOR ||
-	    _system_type == MAV_TYPE_VTOL_TILTROTOR ||
-	    _system_type == MAV_TYPE_VTOL_RESERVED2) {
-
-		/* multirotors: set number of rotor outputs depending on type */
+	unsigned motors_count;
+	bool is_fixed_wing;
 
-		unsigned n;
-
-		switch (_system_type) {
-		case MAV_TYPE_AIRSHIP:
-		case MAV_TYPE_VTOL_DUOROTOR:
-			n = 2;
-			break;
+	switch (_system_type) {
+	case MAV_TYPE_AIRSHIP:
+	case MAV_TYPE_VTOL_DUOROTOR:
+	case MAV_TYPE_COAXIAL:
+		motors_count = 2;
+		is_fixed_wing = false;
+		break;
 
-		case MAV_TYPE_QUADROTOR:
-		case MAV_TYPE_VTOL_QUADROTOR:
-		case MAV_TYPE_VTOL_TILTROTOR:
-			n = 4;
-			break;
+	case MAV_TYPE_TRICOPTER:
+		motors_count = 3;
+		is_fixed_wing = false;
+		break;
 
-		case MAV_TYPE_VTOL_RESERVED2:
-			// this is the standard VTOL / quad plane with 5 propellers
-			n = 5;
-			break;
+	case MAV_TYPE_QUADROTOR:
+	case MAV_TYPE_VTOL_QUADROTOR:
+	case MAV_TYPE_VTOL_TILTROTOR:
+		motors_count = 4;
+		is_fixed_wing = false;
+		break;
 
-		case MAV_TYPE_HEXAROTOR:
-			n = 6;
-			break;
+	case MAV_TYPE_VTOL_RESERVED2:
+		motors_count = 5;
+		is_fixed_wing = false;
+		break;
 
-		default:
-			n = 8;
-			break;
-		}
+	case MAV_TYPE_HEXAROTOR:
+		motors_count = 6;
+		is_fixed_wing = false;
+		break;
 
-		for (unsigned i = 0; i < 16; i++) {
-			if (armed) {
-				if (i < n) {
-					/* scale PWM out PWM_DEFAULT_MIN..PWM_DEFAULT_MAX us to 0..1 for rotors */
-					msg.controls[i] = (_actuator_outputs.output[i] - PWM_DEFAULT_MIN) / (PWM_DEFAULT_MAX - PWM_DEFAULT_MIN);
+	case MAV_TYPE_OCTOROTOR:
+	case MAV_TYPE_SUBMARINE:
+		motors_count = 8;
+		is_fixed_wing = false;
+		break;
 
-				} else {
-					/* scale PWM out PWM_DEFAULT_MIN..PWM_DEFAULT_MAX us to -1..1 for other channels */
-					msg.controls[i] = (_actuator_outputs.output[i] - pwm_center) / ((PWM_DEFAULT_MAX - PWM_DEFAULT_MIN) / 2);
-				}
+	case MAV_TYPE_FIXED_WING:
+		motors_count = 0;
+		is_fixed_wing = true;
+		break;
 
-			} else {
-				/* send 0 when disarmed and for disabled channels */
-				msg.controls[i] = 0.0f;
-			}
-		}
+	default:
+		motors_count = 0;
+		is_fixed_wing = false;
+		break;
+	}
 
-	} else {
-		/* fixed wing: scale throttle to 0..1 and other channels to -1..1 */
+	for (unsigned i = 0; i < actuator_outputs_s::NUM_ACTUATOR_OUTPUTS; i++) {
+		if (!armed) {
+			/* send 0 when disarmed and for disabled channels */
+			msg->controls[i] = 0.0f;
 
-		for (unsigned i = 0; i < 16; i++) {
-			if (armed) {
-				if (i != 4) {
-					/* scale PWM out PWM_DEFAULT_MIN..PWM_DEFAULT_MAX us to -1..1 for normal channels */
-					msg.controls[i] = (_actuator_outputs.output[i] - pwm_center) / ((PWM_DEFAULT_MAX - PWM_DEFAULT_MIN) / 2);
+		} else if ((is_fixed_wing && i == 4) ||
+			   (!is_fixed_wing && i < motors_count)) {	//multirotor, rotor channel
```

Comment:
Could you make the `is_fixed_wing` condition a separate condition so that it is more explicitly handled as a different condition?

Reply:
I think its better how I wrote it.
in that way, the code of doing the scaling for full -1:+1 or 0:+1 written once.